Game

laurieTombola opened the pull request:
Huge pull request. Not sure how to do pull requests when you need everything in it to continue working.

Commit:
Added new images, written the html and less to display the board, and started on the API calls.
Also organised the code into files, modified the concat task to work properly with angularjs.

Commit:
Refactored index.html, added functionality for reseting the game.

Created a base images folder for images that are not to be copied to the build folder.

KITTENS AND PUPPIES ADDED! :D:D:D:D:D:D:D:D:D:D:D:D:D:D

Plenty of functionality added to the controller, including checking for errors before contacting the server.

Commit:
UI Rooting added, directives to shorten / improve the html written, done some CSS to improve the nav bar.

Commit:
last few days of work

Commit:
fixed some grunt tasks  (refactoring code, file names changed, updating grunt)
Moved lots of files around.
Added lots of files that basically are just old files split up.

Commit:
Saving work before OS update

Commit:
Test

Commit:
Lots of refinement and code cleaning.

Commit:
Partial commit of new NavBar controller to enable canceling of timed events when people return to lobby after winning a game but before the code has time to switch to the winner's screen

Also a few bits of code are tidied up.

Commit:
Corrected bad copy pasta.

keithbarrow commented:
43 files. YOU ARE HAVENING\* A TURKISH
- Ref: https://www.chroniclesofgeorge.com/

laurieTombola commented:
Yeah sorry, there was no point at which I could do a pull request without having to stop work till it was sorted.

Commit:
fixed the win controller still running when not in win state
deleted 2 un-needed files
fixed a privacy error in player-type
set up draw state
started working on bootstrapping.

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+    angular.module('Tombola.Games.NoughtsAndCrosses.Game')
+        .controller('Game', ['$scope', '$state', '$interval', 'GameServerProxy', 'PlayerType', 'GameStatus', function($scope, $state, $interval, proxy, playerType, game){
+            var me = this;
+            $scope.pageHeading = "Kittens Vs Puppies! Fighto!";
```
Just hard code in the HTML.

laurieTombola replied:
Its used later to notify players that a game is a win / draw

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+            var me = this;
+            $scope.pageHeading = "Kittens Vs Puppies! Fighto!";
+            $scope.turn = 1;
+            console.log('Game Controller Loaded');
```
remove

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+
+            $scope.clickBox = function(box){
+                if(game.getBoard()[parseInt(box)] === '0' && game.getState() === 'Continue'){
+                    console.log('Box:  '+box+'||| Turn: '+$scope.turn);
```
Remove

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+                            }
+                            game.setState(response.data.outcome);
+                            if(game.getState() === 'Draw'){
+                                console.log('Draw');
```
remove

keithbarrow replied:
remove

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+                                console.log('Draw');
+                            }
+                            else if(game.getState() === 'Win'){
+                                console.log('Win');
```
remove

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+                                console.log('Win');
+                                $scope.pageHeading = 'Player ' + response.data.winner + 'Wins!';
+                                $interval(function(){
+                                    console.log('should be switching to win');
```
remove

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+                        });
+                }
+                else {
+                    //console.log("Already Filled");
```
remove, plus the else condition

keithbarrow commented on main-app/app/scripts/game/controllers/game-controller.js:
```diff
@@ -0,0 +1,87 @@
+                }
+            };
+
+            var isHumanVsComputer = function(){
```
Whazz this into the playerType service the

keithbarrow commented on main-app/app/scripts/game/controllers/noughts-and-crosses-controller.js:
```diff
@@ -0,0 +1,62 @@
+    'use strict';
+
+
+    angular.module('Tombola.Games.NoughtsAndCrosses')
```
Remove

keithbarrow commented on main-app/app/scripts/game/services/make-move.js:
```diff
@@ -0,0 +1,23 @@
+(function () {
+    'use strict';
+
+    angular.module('Tombola.Games.NoughtsAndCrosses.Game')
```
Make Move and new game is better in one service - correctly written the methods have 90% the same code.

keithbarrow commented on main-app/app/scripts/lobby/controllers/lobby-controller.js:
```diff
@@ -0,0 +1,46 @@
+                };
+
+            $scope.showTheGame = function(){
+                if(playerType.validatePlayerType(playerType.getPlayer1().type) && validatePlayerType(playerType.getPlayer2().type) ){
```
Put the validation in the service

keithbarrow commented on main-app/app/scripts/navbar/controller/navbar-controller.js:
```diff
@@ -0,0 +1,11 @@
+(function () {
+    'use strict';
+
+    angular.module('Tombola.Games.NoughtsAndCrosses.Navbar')
```
Does this do anything?

keithbarrow commented:
:shipit: - But fix the problems highlighed in separate branches.

Commit:
Moved some functionality from game-controller to player-type
Removed console logs.
Deleted the navber stuff, it was a failed concept, so it was aborted, its frail life was terminated without it having any say in the matter, it.... is over.

Commit:
Fixed validate player type and references.

laurieTombola committed:
Merge pull request #3 from laurieTombola/Polish

Polish

laurieTombola commented:
Done.
